electron_app/UploadManager.js

```diff
@@ -44,28 +48,33 @@ const uploadManager = (bucketName, fileList = [], access_token, win) => {
             requestList.push(reqConfig);
 
           } else {
-            console.log(JSON.stringify(resp));
-            console.error("There was an error trying to connect to google");
+            console.error("There was an error trying to get location for the following url: ", uri);
+            win.webContents.send(constants.IPC_UPLOAD_FAILED, file);
           }
-
         }
-        progressConf.on('progress', function (progress) {
-          file.progress = Math.round(progress.percentage);
-          file.transferred = progress.transferred;
-          win.webContents.send(constants.IPC_UPLOAD_STATUS, file);
+
+        progressConf.on('progress', (progress) => {
+          if (progress.percentage >= 100.0 && progress.delta === 0) {
+            file.progress = 100;
```

Comment:
There is a case where % progress is > 100?

Reply:
Don't know, Just defensive coding against rounding or similar issues.